annotation/core/models.py

```diff
@@ -12,26 +12,41 @@ def __str__(self):
         return self.user.username + " " + str(self.is_turker)
 
 
-class Tag(models.Model):
-    name = models.CharField(max_length=100)
-    text = models.CharField(max_length=100)
-    human = models.BooleanField()
+class System(models.Model):
+    """The model that generated a given Generation."""
+    name = models.CharField(max_length=200)
+    description = models.TextField()
 
-    def __str__(self):
-        return self.text
+    # @kirubarajan: is this for training data or fine-tuning?
+    finetuned_dataset_location = models.URLField(null=True)
+
+
+class Dataset(models.Model):
+    """Dataset that we sample prompts/true human continuations from."""
+    name = models.CharField(max_length=128)
+    split = models.CharField(max_length=16)
 
 
 class Prompt(models.Model):
     """Human written sentence that gets continued."""
     body = models.TextField()
+    dataset = models.ForeignKey(Dataset, on_delete=models.DO_NOTHING)
 
     def __str__(self):
         return self.body
 
 
-class EvaluationText(models.Model):
+class DecodingStrategy(models.Model):
+    """Meta-data about generation strategy used for a Generation object. """
+    name = models.CharField(max_length=128)
+    value = models.FloatField(null=True)
+
+
+class Generation(models.Model):
     """The continuation associated with the prompt."""
+    system = models.ForeignKey(System, on_delete=models.DO_NOTHING)
     prompt = models.ForeignKey(Prompt, on_delete=models.DO_NOTHING)
+    decoding_strategy = models.ForeignKey(DecodingStrategy, on_delete=models.DO_NOTHING)
     body = models.TextField()
     boundary = models.IntegerField()
```

Comment:
I think this code would be stabler if `boundary` was automatically computed as the number of sentences in the Prompt.

This would prevent any danger of off-by-one errors.